src/systems/filecoin_mining/storage_proving/sealer/sealer.go

```diff
@@ -0,0 +1,57 @@
+package sealer
+
+import file "github.com/filecoin-project/specs/systems/filecoin_files/file"
+import sector "github.com/filecoin-project/specs/systems/filecoin_mining/sector"
+import util "github.com/filecoin-project/specs/util"
+
+func (s *SectorSealer_I) SealSector(si SealInputs) *SectorSealer_SealSector_FunRet_I {
+	sid := si.SectorID()
+
+	commD := sector.UnsealedSectorCID(s.ComputeDataCommitment(si.UnsealedPath()).As_commD())
+
+	buf := make(util.Bytes, si.SealCfg().SectorSize())
+	f := file.FromPath(si.SealedPath())
+	length, _ := f.Read(buf)
+
+	// TODO: How do we meant to handle errors in implementation methods? This could get tedious fast.
+
+	if util.UInt(length) != util.UInt(si.SealCfg().SectorSize()) {
+		panic("Sector file is wrong size.")
+	}
+
+	return &SectorSealer_SealSector_FunRet_I{
+		rawValue: Seal(sid, si.RandomSeed(), commD, buf),
+	}
+}
+
+func (s *SectorSealer_I) VerifySeal(sv sector.SealVerifyInfo) *SectorSealer_VerifySeal_FunRet_I {
+	return &SectorSealer_VerifySeal_FunRet_I{}
+}
+
+func (s *SectorSealer_I) ComputeDataCommitment(unsealedPath file.Path) *SectorSealer_ComputeDataCommitment_FunRet_I {
+	return &SectorSealer_ComputeDataCommitment_FunRet_I{}
+}
+
+func ComputeReplicaID(sid sector.SectorID, commD sector.UnsealedSectorCID, seed sector.SealRandomSeed) *SectorSealer_ComputeReplicaID_FunRet_I {
+
+	_, _ = sid.MinerID(), (sid.Number())
```

Comment:
nit, I'd just add some slight comments where as a spec we want to strike some balance of not having the go-like interface take over potentially more descriptive work...